DirectProgramming/C++SYCL_FPGA/ReferenceDesigns/convolution2d/include/extract_typename.hpp

```diff
@@ -0,0 +1,73 @@
+//  Copyright (c) 2024 Intel Corporation
+//  SPDX-License-Identifier: MIT
+
+// extract_typename.hpp
+
+#pragma once
+
+// C++ magic that lets us extract template parameters from SYCL pipes,
+// `StreamingBeat` structs based off code from https://ideone.com/SDEgq
+
+#include <stdint.h>
+
+#include <sycl/ext/intel/prototype/pipes_ext.hpp>
+
+template <typename T>
+struct ExtractValueType {
+  typedef T value_type;
+};
+
+template <template <typename, typename...> typename X, typename T,
+          typename... Args>
+struct ExtractValueType<X<T, Args...>>  // specialization
+{
+  typedef T value_type;
+};
+
+template <typename T>
+struct ExtractPipeType {
+  typedef T value_type;
+};
+
+template <template <class, class, int32_t, class, typename...> class PIPE_CLASS,
```

Comment:
Why the switch to UPPER_CASE for this types in this one place.   Does not seem consistent with rest of example.